src/marqo/core/unstructured_vespa_index/unstructured_vespa_index.py

```diff
@@ -224,45 +229,146 @@ def _get_score_modifiers(marqo_query: MarqoQuery) -> \
                 else:
                     raise InternalError(f'Unknown score modifier type {modifier.type}')
 
-            # Note one of these could be empty, but not both
-            return {
-                unstructured_common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS: mult_tensor,
-                unstructured_common.QUERY_INPUT_SCORE_MODIFIERS_ADD_WEIGHTS: add_tensor
-            }
+            if self._marqo_index_version < self._VERSION_2_10_0:
+                return {
+                    unstructured_common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS_2_9: mult_tensor,
+                    unstructured_common.QUERY_INPUT_SCORE_MODIFIERS_ADD_WEIGHTS_2_9: add_tensor
+                }
+            elif isinstance(marqo_query, MarqoTensorQuery):
+                return {
+                    unstructured_common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS_TENSOR: mult_tensor,
+                    unstructured_common.QUERY_INPUT_SCORE_MODIFIERS_ADD_WEIGHTS_TENSOR: add_tensor
+                }
+            elif isinstance(marqo_query, MarqoLexicalQuery):
+                return {
+                    unstructured_common.QUERY_INPUT_SCORE_MODIFIERS_MULT_WEIGHTS_LEXICAL: mult_tensor,
+                    unstructured_common.QUERY_INPUT_SCORE_MODIFIERS_ADD_WEIGHTS_LEXICAL: add_tensor
+                }
+            else:
+                raise InternalError(f'Unknown query type {type(marqo_query)}')
 
         return None
 
-    def _to_vespa_lexical_query(self, marqo_query: MarqoLexicalQuery) -> Dict[str, Any]:
-        def _get_lexical_search_term(marqo_query: MarqoLexicalQuery) -> str:
-            if not marqo_query.or_phrases and not marqo_query.and_phrases:
-                return 'false'
-            if marqo_query.or_phrases == ["*"] and not marqo_query.and_phrases:
-                return 'true'
-            
-            if marqo_query.or_phrases and marqo_query.score_modifiers:
-                or_terms = ' OR '.join([
-                    f'default contains "{phrase}"' for phrase in marqo_query.or_phrases
-                ])
-            elif marqo_query.or_phrases and not marqo_query.score_modifiers:
-                or_terms = 'weakAnd(%s)' % ', '.join([
-                    f'default contains "{phrase}"' for phrase in marqo_query.or_phrases
-                ])
-            else:
-                or_terms = ''
-
-            if marqo_query.and_phrases:
-                and_terms = ' AND '.join([
-                    f'default contains "{phrase}"' for phrase in marqo_query.and_phrases
-                ])
-                if or_terms:
-                    or_terms = f'({or_terms})'
-                    and_terms = f' AND ({and_terms})'
+    def _get_hybrid_score_modifiers(self, hybrid_query: MarqoHybridQuery) -> \
```

Comment:
I found a bit of duplication here. Same as the next method. If the logic is exactly same for structured and unstructured. Is it worth extracting them to the base class?

Reply:
I'll extract these to the base class: `_convert_score_modifiers_to_tensors`, `_get_score_modifiers`, `_get_hybrid_score_modifiers`.